index.js

```diff
@@ -4,22 +4,47 @@
  * @param {Object} store - An object representing a single store. See the instructions for details on its shape.
  * @returns {Object[]} The same `stores` array that was inputted.
  */
-function addNewStore(stores, store) {}
+function addNewStore(stores, store) {
+  //stores is an array of objects, store is an object, so stores can use the push method
+  //to add the store at the end of the stores array. The push method is desctructive method.
+  stores.push(store);
+  //return the original array of objects
+  return stores;
+}
 
 /**
  * Removes a store object at the given position.
  * @param {Object[]]} stores - An array of store objects.
  * @param {number} index - A number representing the index of the store to be removed from the array.
  * @returns {Object[]} The same `stores` array that was inputted.
  */
-function removeStoreAtPosition(stores, index) {}
+function removeStoreAtPosition(stores, index) {
+  //splice() method is a destructive method
+  //stores.splice(index,1) means remove one element at the assigned index of the stores array
+  stores.splice(index,1);
+  return stores;
+}
 
 /**
  * Creates a duplicate of the `store` object. No references should be shared between the inputted `store` and the result.
  * @param {Object} store - An object representing a single store. See the instructions for details on its shape.
  * @returns {Object} The duplicated store object. This should not be the same as the store that was inputted.
  */
-function duplicateStore(store) {}
+function duplicateStore(store) {
+  let obj = {}; //create a new object
+  for(let key in store){ //iterate through the object
+    if(Array.isArray(store[key])){ //if the property is an array
+      obj[key] = [].concat(store[key]); //then assign the same array by concat with empty array
```

Comment:
`obj[key] = [...store[key]]` is a more common way to shallow copy this array. The `...` is called the spread operator.